pkg/instrumentation/metrics.go

```diff
@@ -0,0 +1,45 @@
+package instrumentation
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type Metrics struct {
+	repositoriesCreated  prometheus.Counter
+	repositoriesDeleted  prometheus.Counter
+	introspectionSuccess prometheus.Counter
+	introspectionFailure prometheus.Counter
+}
+
+func NewMetrics(reg *prometheus.Registry) *Metrics {
+	metrics := &Metrics{
+		repositoriesCreated: promauto.NewCounter(prometheus.CounterOpts{
+			Name: "repository_create_count",
+			Help: "The number of repositories created",
+		}),
+
+		repositoriesDeleted: promauto.NewCounter(prometheus.CounterOpts{
+			Name: "repository_delete_count",
+			Help: "The number of repositories deleted",
+		}),
+		introspectionSuccess: promauto.NewCounter(prometheus.CounterOpts{
+			Name: "introspection_success_count",
+			Help: "The number of repositories introspected with success",
+		}),
+		introspectionFailure: promauto.NewCounter(prometheus.CounterOpts{
+			Name: "introspection_failure_count",
+			Help: "The number of repositories introspected with failure",
+		}),
+	}
+
+	reg.MustRegister(metrics.introspectionSuccess)
+	reg.MustRegister(metrics.introspectionFailure)
+	reg.MustRegister(metrics.repositoriesCreated)
+	reg.MustRegister(metrics.repositoriesDeleted)
+
+	reg.MustRegister(collectors.NewBuildInfoCollector())
```

Comment:
Playing around with prometheus, I learned that using `promauto` will automatically register the metrics with the `DefaultRegisterer`. In which case, I don't think you need to call `MustRegister` here.

Reply:
You are right about `promauto`! but for any reason that I don't understand yet, when I integrated that into the service, I couldn't get the metric into the response; it was when I used this approach when the metric started to be reported; I don't know the root cause honestly.

Reply:
I have just seen that invokation to `promauto.NewCounter` is invoking `With(prometheus.DefaultRegisterer).NewCounter`; I am doing something wrong here.

Reply:
If you wanted to avoid DefaultRegisterer, I believe you can do `prometheus.NewCounter` instead.